src/puppetlabs/trapperkeeper/core.clj

```diff
@@ -154,33 +157,74 @@
     ;; it can be referenced in a remote nREPL session, etc.
     (swap! internal/tk-apps conj app)
     (internal/register-sighup-handler)
-    (run-app app)
-    (swap! internal/tk-apps (partial remove #{app}))))
+    (let [{:keys [::exit] :as result} (run-app app)]
+      ;; Q: If it's appropriate (or even just acceptable) for this
+      ;; removal to be handled via catch/finally, then it'd be a bit
+      ;; simpler to just throw the exit from run-app.
```

Comment:
Not sure about the answer here. Though throwing from `run-app` is consistent with the throw we're already doing for for errors, which could be nice.

Reply:
I'm into simplicity, so if throwing from `run-app` is simpler..

Reply:
also curious though: why is the exit being `throw`n anyway? does that not just cause an error? I thought these were intentional exits?

Reply:
@Magisus agreed, and to elaborate, my concern was really just whether there might be some reason TK really meant to avoid the removal on exception -- but I could also imagine it was just an oversight, and that it should always be removed.  If we can convince ourselves that the removal is "fine", then I'd be inclined to switch it to a finally or similar.

Reply:
@mwaggett the `::exit` throws are intentional exits -- see the corresponding catch in `main` below.  We do that so that you can exit "immediately", while still having an opportunity to run any pending handlers, i.e. `finally`, `catch`, etc.

Reply:
ahh gotcha, that makes sense, thanks!